src/hooks/usePositionFixed.ts

```diff
@@ -1,45 +1,50 @@
 /** Position fixed breaks in mobile Safari when the keyboard is up. This module provides functionality to emulate position:fixed by changing all top navigation to position:absolute and updating on scroll. */
-import { useEffect } from 'react'
 import { token } from '../../styled-system/tokens'
-import { isIOS, isSafari, isTouch } from '../browser'
-import * as selection from '../device/selection'
-import reactMinistore from '../stores/react-ministore'
-import once from '../util/once'
+import safariKeyboardStore from '../stores/safariKeyboardStore'
+import viewportStore from '../stores/viewport'
 import useScrollTop from './useScrollTop'
 
-/** Default mode is fixed. Absolute mode emulates positioned fixed by switching to position absolute and updating the y position on scroll. */
-const positionFixedStore = reactMinistore<'fixed' | 'absolute' | null>(null)
-
-/** Initializes handler to switch position fixed elements to position absolute when the keyboard is up on mobile Safari. Initialized once, permanently since the platform cannot change. */
-const initEventHandler = once(() => {
-  /** Updates the positionFixedStore state based on if the keyboard is visible. */
-  const updatePositionFixed = () => {
-    const keyboardIsVisible = selection.isActive()
-    positionFixedStore.update(keyboardIsVisible ? 'absolute' : 'fixed')
-  }
-
-  if (isTouch && isSafari() && !isIOS) {
-    updatePositionFixed()
-    document.addEventListener('selectionchange', updatePositionFixed)
-  }
-})
-
 /** Emulates position fixed on mobile Safari with positon absolute. Returns { position, overflowX, top } in absolute mode. */
-const usePositionFixed = (): {
+const usePositionFixed = ({
+  fromBottom,
+  offset = 0,
+  height,
+}: {
+  fromBottom?: boolean
+  offset?: number
+  /** Only for if `fromBottom = true`. For calculating position on mobile safari. */
+  height?: number
+} = {}): {
   position: 'fixed' | 'absolute'
   overflowX?: 'hidden' | 'visible'
-  top: string
+  top?: string
+  bottom?: string
+  display?: 'none'
 } => {
-  const position = positionFixedStore.useState()
+  const safariKeyboard = safariKeyboardStore.useState()
+  const position = safariKeyboard.open ? 'absolute' : 'fixed'
   const scrollTop = useScrollTop({ disabled: position === 'fixed' })
+  const { innerHeight, currentKeyboardHeight } = viewportStore.useState()
 
-  useEffect(initEventHandler, [])
+  let top, bottom
+  if (position === 'absolute') {
+    top = fromBottom
+      ? `${scrollTop + innerHeight - currentKeyboardHeight - (height ?? 0) - offset}px`
+      : `${scrollTop + offset}px`
+  } else if (fromBottom) {
+    bottom = `calc(${token('spacing.safeAreaBottom')} + ${offset + currentKeyboardHeight}px)`
```

Comment:
Wouldn't `currentKeyboardHeight` always be 0 in this case, since the keyboard is closed?

Reply:
Yep! Refactored that one out here: 99419f66bf875679e633a2322e1cc819e1873560